### PR TITLE
feat: add `apt`

### LINKED ISSUE #1397: Add `apt` completion spec
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [ ] I would like to work on this.

### What CLI tool does this relate to?

apt

### Which statement makes the most sense?

This completion spec does not exist

### Issue Details

Also `apt-get`
https://en.wikipedia.org/wiki/APT_(software)

### PR DESCRIPTION
**Pull request by mrhappyma**
potentially closes #1397 

big thanks to Grant and Robert for helping me get generators figured out :)

### CODE REVIEW

**Review by SeparateRecords — COMMENTED** [review]
I know it's still in progress but this is looking really great! I don't have a VM set up to test this so will leave that to @grant0417 :)

File: src/apt.ts [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    }
+    // Only lines matching the first character, delete characters after '/'
+    const out = await executeShellCommand(
+      `apt list | grep '^${finalToken[0]}' | sed 's#/.*##g'`
+    );
+    return out.split("\n").map((name) => ({
+      name,
+      description: "Package",
+      icon: "📦",
+      priority: 50,
```
[the remark this answers is not in the document]
Comment (mrhappyma) [reply src/apt.ts:22]: Without it the folders from the `filepaths` template seemed to have higher priority than the generator and hide it's suggestions entirely. Would there be better way of fixing this?

[the remark this answers is not in the document]
Comment (SeparateRecords) [reply src/apt.ts:22]: Ohhh that makes sense. Honestly the whole priority algorithm is still a black box to sometimes it does some weird stuff like that. 

I'd say keep the line, but document why it's there in a comment above it

File: src/apt.ts [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    name: "-y",
+    description: "Assume yes to all prompts",
+  },
+  {
+    name: "--assume-no",
+    description: "Assume no to all prompts",
+  },
+];
+
+const instillationOptions: Fig.Option[] = [
```
Comment (fedeci) [inline src/apt.ts:72]: typo:
```suggestion
const installationOptions: Fig.Option[] = [
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks, I cannot test generators but the spec looks good overall!

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Apparently forgot to submit the review. Should be easy changes

File: src/apt.ts [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+        ...yesNoOptions,
+        ...simulate,
+        {
+          name: "--compile",
+          description:
+            "Compile the package to a binary using dpkg-buildpackage",
+        },
+        {
+          name: "--only-source",
+          // no idea how this works
```
[the remark this answers is not in the document]
Comment (mrhappyma) [reply src/apt.ts:262]: phew, when I saw that there I was worried you were going to ask for a description

File: src/apt.ts [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+        isVariadic: true,
+      },
+      options: [...instillationOptions, ...yesNoOptions],
+    },
+  ],
+  options: [
+    {
+      name: ["-h", "--help"],
+      description: "Print help message and exit",
+    },
```
[the remark this answers is not in the document]
Comment (mrhappyma) [reply src/apt.ts:333]: I was debating about this, while it does technically work with all subcommands, the help message doesn't change. Should this still appear?

File: src/apt.ts [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+    name: "-y",
+    description: "Assume yes to all prompts",
+  },
+  {
+    name: "--assume-no",
+    description: "Assume no to all prompts",
+  },
+];
+
+const instillationOptions: Fig.Option[] = [
```
[the remark this answers is not in the document]
Comment (mrhappyma) [reply src/apt.ts:72]: :+1: fixed this locally and just waiting for the linter to do its thing

File: src/apt.ts [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+const yesNoOptions: Fig.Option[] = [
+  {
+    name: "-y",
+    description: "Assume yes to all prompts",
+  },
+  {
+    name: "--assume-no",
+    description: "Assume no to all prompts",
+  },
+];
```
[the remark this answers is not in the document]
Comment (SeparateRecords) [reply src/apt.ts:70]: to clarify, should have `exclusiveOn: ["--assume-no"]` on `-y` and `exclusiveOn: ["-y"]` on `--assume-no`

File: src/apt.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (SeparateRecords) [reply src/apt.ts:336]: I'd say so! If it's valid then may as well, can de-prioritize it (`priority: 40`)

[the remark this answers is not in the document]
Comment (mrhappyma) [reply src/apt.ts]: Done!
I've done that to the version option too, as it works similarly.

**Comment by mrhappyma** [comment]
I give up, re-requesting review isn't working.

**Review by SeparateRecords — APPROVED** [review]
LGTM! Thank you @mrhappyma! 😊

apt is particular thorny because of the lack of docs, really appreciate the effort you went to to make sure things are as correct as they can realistically be.

**Pull request merged**